The listener_table can be NULL if we have started connman with -r
option. Because we return <0 value to caller in this case,
the caller (like tethering code) has possibility to change how
its functionality (fall back to some other DNS servers).
---
 src/dnsproxy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dnsproxy.c b/src/dnsproxy.c
index 52d67d4..8ed34a7 100644
--- a/src/dnsproxy.c
+++ b/src/dnsproxy.c
@@ -3426,7 +3426,7 @@ int __connman_dnsproxy_add_listener(int index)
 		return -EINVAL;
 
 	if (listener_table == NULL)
-		return 0;
+		return -ENOENT;
 
 	if (g_hash_table_lookup(listener_table, GINT_TO_POINTER(index)) != NULL)
 		return 0;
-- 1.7.11.7

